protocol.html

```diff
@@ -671,8 +681,8 @@ <h3 property="schema:name">Description Resource</h3>
 
                   <ul>
                     <li id="notify-resource-id">One <code>id</code> property to identify the resource described by the description resource.</li>
-                    <li id="notify-subscription">Zero, one or many <code rel="dcterms:subject" resource="http://www.w3.org/ns/solid/notifications#subscription">subscription</code> property to identify the available <a href="#subscription-resource">subscription resources</a>.</li>
-                    <li id="notify-channel">Zero, one or many <code rel="dcterms:subject" resource="http://www.w3.org/ns/solid/notifications#channel">channel</code> property to identify the available <a href="#notification-channel">notification channels</a>.</li>
+                    <li id="notify-subscription">Zero or one <code rel="dcterms:subject" resource="http://www.w3.org/ns/solid/notifications#subscription">subscription</code> property with the property value containing zero, one or many entries to specify the available <a href="#subscription-resource">subscription resources</a>.</li>
```

Comment:
I see the bulk of your objections relate to changes in the data model. My goal here was to use very precise language as we are *specifying*. 

For example, in the text above, we cannot add multiple `subscription` property to the data model. We can add at most one `subscription` which contains a list of subscriptions. I do not think the original text adequately captures this.

I understand that this sort of thing makes specifications hard to read. But I am sure you agree that preciseness is the purpose of specifications. Perhaps, we need to discuss this further.

Reply:
The original text indeed allows zero, one, or more subscription properties! That is, in the description resource, one can discover multiple subscription and channels to be available.

In the Turtle, that'd be expressed like:

```turtle
<>
  notify:subscription <https://websocket.example/subscription>,
                      <https://ldn.example/subscription/>,
                      <https://streaminghttp.example/subscription> .
# the above is equivalent to writing:
# notify:subscription <https://websocket.example/subscription> .
# notify:subscription <https://ldn.example/subscription/> .
# notify:subscription <https://streaminghttp.example/subscription> .

<https://websocket.example/subscription>
  notify:channelType notify:WebSocketChannel2023 .

<https://ldn.example/subscription/>
  notify:channelType notify:LDNChannel2023 .

<https://streaminghttp.example/subscription>
  notify:channelType notify:StreamingHTTPChannel2023 .
```

Again, the description need not be specific about the encoding but just focus on cardinality. It is only the example that shows how it is codified in JSON-LD.

Reply:
I understand where you are coming from, but I have to disagree here. What you have cited as the example shows one property/predicate that has multiple values/objects. 

My contention is that we want to expressly prohibit (and make client life easier):
```json
{
  "subscription": [...],
  "subscription": [...]
} 
```

I am afraid the current langauge does not do this! I am not at all arguing that we focus on encoding. But in talking about cardinality, we must explicitly capture if we are talking about the cardinality for the predicate/property or object/property value. 

---

I guess my larger point is that I find the use of "property" and "values" in this section confusing. For example, for "features" you say:

> Zero, one or many feature **property value(s)** to state the supported features

But to follow you reasoning, we could write:

> Zero, one or many feature **properties** to state the supported features.

But obviously this is confusing, because in the notification-channel-data-model we describe feature:modifier pairs as

> Zero, one or many properties stating a particular feature

I ask that we be careful here because those using the spec may not have your knowledge. It is helpful to be slightly ignorant here; catering to that ignorance will lead to a better spec.